config/src/main/java/com/typesafe/config/impl/ConfigNodeSimpleValue.java

```diff
@@ -16,4 +17,21 @@
     protected Collection<Token> tokens() {
         return Collections.singletonList(token);
     }
+
+    protected Token token() { return token; }
+
+    protected AbstractConfigValue value() {
+        if (Tokens.isValue(token))
+            return Tokens.getValue(token);
+        else if (Tokens.isUnquotedText(token))
+            return new ConfigString.Unquoted(token.origin(), Tokens.getUnquotedText(token));
+        else if (Tokens.isSubstitution(token)) {
+            List<Token> expression = Tokens.getSubstitutionPathExpression(token);
+            Path path = PathParser.parsePathExpression(expression.iterator(), token.origin());
+            boolean optional = Tokens.getSubstitutionOptional(token);
+
+            return new ConfigReference(token.origin(), new SubstitutionExpression(path, optional));
+        }
+        return null;
```

Comment:
I'm suspicious of this null - when does it happen? Often I'd call a method that can return null `valueOrNull` for clarity, but depending on when it happens I don't know if it should be an exception instead, or what.